routes/api/stories.js

```diff
@@ -0,0 +1,48 @@
+const express = require('express');
+const {Post, User,Follow} = require('../../db/models');
+const {getRandomInt} = require('../utils.js')
+const router = express.Router();
+const {restoreUser} = require('../../auth.js');
+const limit = 6;
+const { Op } = require("sequelize");
+router.get('/stories', async (req,res,next) => { 
+    console.log(req.session.offset)
+    try{
+        // TODO write a query to get 6 random rows
+        const getStories = await Post.findAll({
+            include: User,
+            limit,
+            order:[['createdAt','DESC']],
+            offset:req.session.offset
+        });
+        req.session.offset += 6;
+        res.json(getStories)
+    }catch(e){
+        next(e);
+    }
+});
+//TODO route to get people followed and their posts
+
+router.get('/stories/following',restoreUser, async(req,res,next) => {
+    try{
+        const getFollowing = await Post.findAll({
+            include:{
+                model: User,
+                include:{
+                    model:User,
+                    as:"Followers",
+                    where:{
+                        id:res.locals.user.id
+                    }
+                }
+            },
+        })
+        let getPosts = getFollowing.filter(e => {
+            if(e.User) return true;
+        })
```

Comment:
Can `e.User` be undefined here?

Reply:
The query returns what I want, but for some reason its still including posts of people that the logged in user does not follow, but with the user set to null. It's just filtering out those null values so I get only the posts that the user follows.

Reply:
Ok sounds good. I'm sure there's a way to optimize the Sequelize query to get exactly what is desired, but as long as this is working it's fine.